### DISCUSSION TITLE
eglot-code-action-organize-imports does not seem to work

### DISCUSSION THREAD

**Discussion by zhaozhemin** [joaotavora/eglot#1002] [Q&A]
I have a .ts file:

    import {sub1} from './sub1'
    import {add1} from './add1'

    add1(1) + sub1(1)

I would expect that the second line would appear before the first line after running `eglot-code-action-organize-imports`, but instead it only prints a message saying 'No "source.organizeImports" code actions here'.

Below is the relevant part of the events buffer:

    [client-request] (id:220) Wed Jul 27 15:42:00 2022:
    (:jsonrpc "2.0" :id 220 :method "textDocument/codeAction" :params
              (:textDocument
               (:uri "file:///private/tmp/react-playground/src/utils.ts")
               :range
               (:start
                (:line 4 :character 0)
                :end
                (:line 4 :character 0))
               :context
               (:diagnostics
                []
                :only
                ["source.organizeImports"])))
    [server-reply] (id:220) Wed Jul 27 15:42:00 2022:
    (:jsonrpc "2.0" :id 220 :result
              [(:title "Organize imports" :edit
                       (:documentChanges
                        [(:textDocument
                          (:uri "file:///private/tmp/react-playground/src/utils.ts" :version 111)
                          :edits
                          [(:range
                            (:start
                             (:line 0 :character 0)
                             :end
                             (:line 1 :character 0))
                            :newText "import { add1 } from './add1'\nimport { sub1 } from './sub1'\n")
                           (:range
                            (:start
                             (:line 1 :character 0)
                             :end
                             (:line 2 :character 0))
                            :newText "")])])
                       :kind "source.organizeImports.ts")])

It seems that the server replies with organized imports, but Eglot doesn't do anything about it.

Is this a bug or am I doing something wrong?

Thanks.

* * *

- OS: macOS Catalina 10.15.7
- Emacs: 28.1
- Eglot: 20220726.1405

**Comment by zhaozhemin** [comment]
I kinda figured it out. Notice that the client sent `source.organizeImports` but the server replied with `source.organizeImports.ts`. The mismatch seems to be the culprit here. You can override the default action or add a new one by calling `(eglot--code-action eglot-code-action-organize-imports "source.organizeImports.ts")` in `typescript-mode-hook`. Now it works correctly.

**Reply by creack** [reply]
Using emacs28.2, this solution doesn't seem to work. Looking at the logs, indeed, the server replies with `source.organizeImports.ts`. I tried this, and also tried to defined a different action, just in case, but I couldn't get eglot to organize the imports.
Any suggestions? Not sure how to troubleshoot.

**Reply by 6b6279** [reply]
I am not sure if this is an issue on `eglot`'s side. Is the server using non-standard replies?
